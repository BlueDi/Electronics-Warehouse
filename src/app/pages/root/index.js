import React, { Component } from 'react';
import Helmet from 'react-helmet';
import { hot } from 'react-hot-loader';
import { renderRoutes } from 'react-router-config';
import { withCookies } from 'react-cookie';
import { Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import { HELMET } from '@config';
import { routes } from '@app/routes';
import { WHHeader } from '@pages';
import '@common/styles/global.css';
import { Grid } from 'semantic-ui-react';

const options = {
  type: 'success',
  position: 'bottom center',
  timeout: 3000,
  offset: '5px'
};

const body_style = {
  height: '90%',
  paddingLeft: '2%',
  paddingRight: '2%'
};

const Layout = () => (
  <Grid stretched style={{ height: '100%' }}>
    <Grid.Row style={{ height: '10%' }}>
      <Grid.Column>
        <WHHeader />
      </Grid.Column>
    </Grid.Row>
    <Grid.Row style={body_style}>
      <Grid.Column>{renderRoutes(routes)}</Grid.Column>
    </Grid.Row>
  </Grid>
);

/**
 * Base of the app.
 * First component to be called.
 */
class Root extends Component {
  render() {
    return (
      <AlertProvider template={AlertTemplate} {...options}>
        <Helmet {...HELMET} />
        <Layout />
      </AlertProvider>
    );
  }
}

export { Layout };
export default hot(module)(withCookies(Root));
