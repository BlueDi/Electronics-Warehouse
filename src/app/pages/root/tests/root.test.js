import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import { Layout } from '@pages/root';

describe('Layout', () => {
  it('should be defined', () => {
    expect(Layout).toBeDefined();
  });

  it('renders properly', () => {
    const component = renderer.create(
      <Router>
        <Layout />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
