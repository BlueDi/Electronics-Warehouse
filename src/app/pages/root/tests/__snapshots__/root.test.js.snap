// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders properly 1`] = `
<div
  className="ui stretched grid"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="row"
    style={
      Object {
        "height": "10%",
      }
    }
  >
    <div
      className="column"
    >
      <div
        className="ui active transition visible inverted dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui indeterminate inverted loader"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "height": "90%",
        "paddingLeft": "2%",
        "paddingRight": "2%",
      }
    }
  >
    <div
      className="column"
    >
      <div
        className="ui active transition visible inverted dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui indeterminate inverted loader"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
